src/Refine/Flag_Check.cpp

```diff
@@ -338,3 +353,60 @@ bool Check_Curl( const int i, const int j, const int k,
    return Flag;
 
 } // FUNCTION : Check_Curl
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Check_Angular
+// Description :  Check if dh/R at cell (i,j,k) is larger than the minimum angular resolution and smaller
+//                than the maximum angular resolution. Also, check if the cell is in Refined_Rad.
+//
+// Note        :  1. Enabled by the runtime option "OPT__FLAG_ANGULAR"
+//
+// Parameter   :  i,j,k       : Indices of the target element in the patch ptr[0][lv][PID]
+//                lv          : Refinement level of the target patch
+//                PID         : ID of the target patch
+//                CenX        : x coordinate of center of calculating angular resolution
+//                CenY        : y coordinate of center of calculating angular resolution
+//                CenZ        : z coordinate of center of calculating angular resolution
+//                AngRes_Max  : the maximum allowed angular resolution
+//                AngRes_Min  : the minimum allowed angular resolution
+//                Refine_Rad  : the radius of must refined at level (lv)
+//
+// Return      :  0 : if the cell is not in the refine region.
+//                1 : if the minimum angular resolution is     reached or  r <  Refine_Rad
+//                2 : if the minimum angular resolution is not reached and r >= Refine_Rad
+//-------------------------------------------------------------------------------------------------------
+int Check_Angular( const int i, const int j, const int k, const int lv, const int PID,
+                   const real CenX, const real CenY, const real CenZ,
+                   const real AngRes_Max, const real AngRes_Min, const real Refine_Rad )
+{
+
+// check
+#  ifdef GAMER_DEBUG
+   if (  i < 0  ||  i >= PS1  ||  j < 0  ||  j >= PS1  ||  k < 0  ||  k >= PS1  )
+      Aux_Error( ERROR_INFO, "incorrect index (i,j,k) = (%d,%d,%d) !!\n", i, j, k );
+#  endif // #ifdef GAMER_DEBUG
+
+   const double dh     = amr->dh[lv];                                         // cell size
+   const double Pos[3] = { amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh,     // x,y,z position
+                           amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh,
+                           amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh  };
+   const double dR [3] = { Pos[0]-CenX, Pos[1]-CenY, Pos[2]-CenZ };
+   const double R      = sqrt( SQR(dR[0]) + SQR(dR[1]) + SQR(dR[2]) );
+
+// (1) check the maximum allowed refinement level based on angular resolution
+   if ( AngRes_Max > 0.0  &&  2.0 * R * AngRes_Max > dh )
+      return 0;
+
+// (2) refine the region within r < Refine_Rad and the innermost cells
+   if ( R < dh  ||  R * UNIT_L < Refine_Rad )
+      return 1;
+
+// (3) check if the minimum angular resolution is reached
+   if ( AngRes_Min > 0.0  &&  R * AngRes_Min < dh )
+      return 1;
```

Comment:
- I think the Criterion (2) should be listed first if the priority of `Refine_Rad` is higher than the angular resolution. 
- It seems that we assume the input `Refine_Rad` is in some units that are not mentioned in `Input__Flag_AngularResolution`.  Should we force the unit of `Refine_Rad` to be code unit, or convert its value to in code unit, in order to save the time for unit conversion (`R * UNIT_L < Refine_Rad`)?
- In the document we should note the users that the priority of `AngRes_Max` is higher than `AngRes_Min` in case of `2.0 * AngRes_Max > AngRes_Min`.

Reply:
`Refine_Rad` should be in the code unit like most of the built-in parameters. Remove `UNIT_L`.

Reply:
It appears to be a bug in the `ccsn` branch that we overlooked.